NPE when retrofit used by IntelliJ IDEA plugin

What kind of issue is this?

 - [ ] Question. This issue tracker is not the place for questions. If you want to ask how to do
       something, or to understand why something isn't working the way you expect it to, use Stack
       Overflow. https://stackoverflow.com/questions/tagged/retrofit

 - [X ] Bug report. If you’ve found a bug, spend the time to write a failing test. Bugs with tests
       get fixed. Here’s an example: https://gist.github.com/swankjesse/6608b4713ad80988cdc9

 - [ ] Feature Request. Start by telling us what problem you’re trying to solve. Often a solution
       already exists! Don’t send pull requests to implement new features without first getting our
       support. Sometimes we leave features out on purpose to keep the project small.

We are developing a plugin for IntelliJ IDEA Ultimate. The plugin uses retrofit for communication with a service. The plugin is crashing due to a NullPointerException resulting from retrofit's platform detection getting confused by the presence of an Android class on the IntelliJ IDEA classpath. The class is present due to IntelliJ's bundled Android Support plugin. The stack trace is:
```
2016-11-30 14:30:27,654 [   7389]  ERROR - plication.impl.ApplicationImpl - null
java.lang.ExceptionInInitializerError
        at java.lang.Class.forName0(Native Method)
        at java.lang.Class.forName(Class.java:264)
        at retrofit2.Platform.findPlatform(Platform.java:37)
        at retrofit2.Platform.<clinit>(Platform.java:29)
        at retrofit2.Retrofit$Builder.<init>(Retrofit.java:402)
        ...
        at com.intellij.openapi.application.impl.ApplicationImpl$3.call(ApplicationImpl.java:331)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
        at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.NullPointerException
        at android.os.SystemProperties_Delegate.native_get(SystemProperties_Delegate.java:44)
        at android.os.SystemProperties.native_get(SystemProperties.java)
        at android.os.SystemProperties.get(SystemProperties.java:64)
        at android.os.Build.getString(Build.java:819)
        at android.os.Build.<clinit>(Build.java:39)
        ... 13 more
```
The relevant portion of the retrofit code is from the Platform class:
```java
  private static Platform findPlatform() {
    try {
      Class.forName("android.os.Build");
      if (Build.VERSION.SDK_INT != 0) {
        return new Android();
      }
    } catch (ClassNotFoundException ignored) {
    }
    try {
      Class.forName("java.util.Optional");
      return new Java8();
    } catch (ClassNotFoundException ignored) {
    }
    return new Platform();
  }
```
The environment is RedHat Linux 6.6 running IntelliJ IDEA Ultimate 2016.3. That version of IntelliJ comes bundled with the Android Support plugin version 10.2.2.

--- a/retrofit/android-test/src/androidTest/java/retrofit2/PlatformTest.java
+++ b/retrofit/android-test/src/androidTest/java/retrofit2/PlatformTest.java
@@ -1,0 +1,26 @@
+/*
+ * Copyright (C) 2020 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package retrofit2;
+
+import org.junit.Test;
+
+import static org.junit.Assert.assertTrue;
+
+public final class PlatformTest {
+  @Test public void isAndroid() {
+    assertTrue(Platform.get() instanceof Platform.Android);
+  }
+}

--- a/retrofit/src/main/java/retrofit2/Platform.java
+++ b/retrofit/src/main/java/retrofit2/Platform.java
@@ -39,14 +39,9 @@ class Platform {
   }
 
   private static Platform findPlatform() {
-    try {
-      Class.forName("android.os.Build");
-      if (Build.VERSION.SDK_INT != 0) {
-        return new Android();
-      }
-    } catch (ClassNotFoundException ignored) {
-    }
-    return new Platform(true);
+    return "Dalvik".equals(System.getProperty("java.vm.name"))
+        ? new Android() //
+        : new Platform(true);
   }
 
   private final boolean hasJava8Types;

--- a/retrofit/src/test/java/retrofit2/PlatformTest.java
+++ b/retrofit/src/test/java/retrofit2/PlatformTest.java
@@ -1,0 +1,27 @@
+/*
+ * Copyright (C) 2020 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package retrofit2;
+
+import static org.junit.Assert.assertFalse;
+
+import org.junit.Test;
+
+public final class PlatformTest {
+  @Test
+  public void isAndroid() {
+    assertFalse(Platform.get() instanceof Platform.Android);
+  }
+}

